### PR TITLE
conmon: store open FDs and close only them

### PR DESCRIPTION
**Pull request by giuseppe**
now that we use a delay to call the cleanup program, we might end up in a race where the event fd used by glib is close'd and it causes the glib event handler to keep polling the closed file descriptor in a tight loop.
    
To avoid closing files that are handled by glib, store what FDs are
opened when conmon first started and close only them.
    
Closes: https://github.com/containers/conmon/issues/221
    
Signed-off-by: Giuseppe Scrivano <gscrivan@redhat.com>

### CODE REVIEW

File: src/conmon.c [hunk 1]
```diff
@@ -22,6 +22,53 @@
 #include <sys/prctl.h>
 #include <sys/stat.h>
 
+static int open_files_max_fd;
```
Comment (haircommander) [inline src/conmon.c:25]: can these additions live in a different file?

File: src/conmon.c [hunk 2]
```diff
@@ -464,24 +512,14 @@ int main(int argc, char *argv[])
...
-		closedir(fdsdir);
-	}
+	int fd;
+
+	for (fd = 3; fd < open_files_max_fd; fd++)
+		if (open_files_set == NULL || FD_ISSET (fd % FD_SETSIZE, &(open_files_set[fd / FD_SETSIZE])))
+			if (fd == sync_pipe_fd || fd == attach_pipe_fd || fd == dev_null_r || fd == dev_null_w
+			    || fd == oom_cgroup_fd || fd == oom_event_fd)
+				close (fd);
+	close_all_readers();
```
Comment (haircommander) [inline src/conmon.c:522]: and this could all also live in that new file

File: src/conmon.c (hunk 1 above)
Comment (giuseppe) [reply src/conmon.c]: sure, moved to a new file

File: src/conmon.c (hunk 2 above)
Comment (giuseppe) [reply src/conmon.c]: same

**Comment by haircommander** [comment]
LGTM

**Comment by rhatdan** [comment]
LGTM
@giuseppe Cleanup format and then we can get this merged and released.

**Comment by giuseppe** [comment]
fixed clang-format errors

**Pull request merged**